### PR TITLE
Fix: Add class options for `lines-around-comment` (fixes #8564)

### LINKED ISSUE #8564: lines-around-comment ES2015 classes in new allowClassStart option
**Tell us about your environment**

* **ESLint Version:** v4.0.0-alpha.2
* **Node Version:** v7.10.0
* **npm Version:** 4.5.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
`--env es6 --rule lines-around-comment:["error",{allowObjectStart:true}]`

**What did you do? Please include the actual source code causing the issue.**

```js
class A {
  /* test */
  constructor() {}
}
```

**What did you expect to happen?**
No errors.

**What actually happened? Please include the actual, raw output from ESLint.**
  2:3  error  Expected line before comment  lines-around-comment

For #2894 @gyandeeps added ES2015 classes to "allowBlockStart" via 9a2dcc2 disallowing an empty line before a comment block just within a class, but it seems like it would be more appropriate as part of its own "allowClassStart".

This proposed change would better match the breaking change in #7879 by @alberto in e232464 where classes are treated differently than block statements. In particular, if someone had `--rule padded-blocks:["error","never"]` in addition to the above `--rule lines-around-comment:["error",{allowObjectStart:true}]`, neither rule would be happy with a line or without a line, but if the code was using `let A = {` instead of a `class`, it would be fine.

~~I would think that moving class from `allowBlockStart` would also be a breaking change.~~

### PR DESCRIPTION
**Pull request by Mardak**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Fix #8564.

**What changes did you make? (Give an overview)**
~~Moved the "ClassBody" check from "Block" to "Object"~~ Added `allowClassStart` and `allowClassEnd` options, updated tests, and added docs.

**Is there anything you'd like reviewers to focus on?**
~~Should this be marked breaking?~~

### CODE REVIEW

**Comment by Mardak** [comment]
This PR should no longer be a breaking rule change as indicated by the old tag.

File: docs/rules/lines-around-comment.md [hunk 1]
```diff
@@ -172,6 +172,55 @@ function foo(){
 }
 ```
 
+### allowClassStart
+
+Examples of **correct** code for this rule with the `{ "beforeLineComment": true, "allowClassStart": true }` option:
```
Comment (ilyavolodin) [inline docs/rules/lines-around-comment.md]: Can you add a few incorrect code examples too. For example, `{ "beforeLineComment": true, "allowClassStart": false }`, or `{ "beforeLineComment": false, "allowClassStart": true }`

Comment (Mardak) [inline docs/rules/lines-around-comment.md]: Added incorrect examples with `allowClassStart: false`. It's not as relevant for incorrect examples with `allowClassStart: true` as it's making the check less strict.

**Comment by Mardak** [comment]
Do the labels/tags on this PR need to be adjusted?

**Comment by kaicataldo** [comment]
This has now been accepted. Apologies for the delay!

File: docs/rules/lines-around-comment.md [hunk 2]
```diff
@@ -172,6 +172,101 @@ function foo(){
...
+### allowClassStart
+
+Examples of **incorrect** code for this rule with the `{ "beforeLineComment": true, "allowClassStart": false }` option:
+
+```js
+/*eslint lines-around-comment: ["error", { "beforeLineComment": true, "allowClassStart": false }]*/
+
+class foo {
+    // what a great and wonderful day
+    day() {"great"}
```
Comment (kaicataldo) [inline docs/rules/lines-around-comment.md]: Can we all these examples say `day() {}` or `day() { return great; }` instead? It's a bit confusing as it currently stands.

Comment (Mardak) [reply docs/rules/lines-around-comment.md]: Switched them all to `day() {}` (and rebased)

**Comment by kaicataldo** [comment]
LGTM. Thank for contributing to ESLint!

**Comment by mysticatea** [comment]
Just FYI, [the release note](https://eslint.org/blog/2017/10/eslint-v4.9.0-released) doesn't seem to list this PR in the Enhancements section.

**Comment by kaicataldo** [comment]
This was my fault. I missed that the commit message said `Fix` at the beginning...

**Comment by kaicataldo** [comment]
Relevant comment: https://github.com/eslint/eslint/issues/8633#issuecomment-336908394

**Pull request merged**